WindowsCompatibility/WindowsCompatibility.psm1

```diff
@@ -695,14 +695,24 @@ function Add-WindowsPSModulePath
         return
     }
 
-    $WindowsPSModulePath = [System.Environment]::GetEnvironmentVariable("psmodulepath", [System.EnvironmentVariableTarget]::Machine)
-    if (-not ($env:PSModulePath).Contains($WindowsPSModulePath))
+    $WindowsPSModulePath = [System.Environment]::GetEnvironmentVariable("PSModulePath", [System.EnvironmentVariableTarget]::Machine)
+    if (-not $env:PSModulePath.Contains($WindowsPSModulePath))
```

Comment:
The order matters. if you look at the original code it was `";${env:userprofile}\Documents\WindowsPowerShell\Modules;${env:programfiles}\WindowsPowerShell\Modules;${WindowsPSModulePath}"`

1. `${env:userprofile}\Documents\WindowsPowerShell\Modules`
1. `${env:programfiles}\WindowsPowerShell\Modules`
1. `$WindowsPSModulePath`

Your code here is in reverse order.

Reply:
Yes, the order matters indeed. And the `$WindowsPSModulePath` should be the first check, because if we would use it after `${env:userprofile}` and `${env:programfiles}` checks and `$WindowsPSModulePath` contains some non-default paths, for example (splitted for readability):
```
C:\Program Files\WindowsPowerShell\Modules;
C:\MyVeryOwnModules;
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
```
then results again will contain duplicate entries (splitted for readability)
```
C:\Users\Username\Documents\PowerShell\Modules;
C:\Program Files\PowerShell\Modules;
c:\program files\powershell\6.0.2\Modules;
C:\Users\Username\Documents\WindowsPowerShell\Modules;
C:\Program Files\WindowsPowerShell\Modules;
C:\Program Files\WindowsPowerShell\Modules;
C:\MyVeryOwnModules;
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
```
because we would first add parts of the `$WindowsPSModulePath` and then, after `$WindowsPSModulePath` check it will be added entirely.

With the order I proposed, the results contain only unique entries (splitted for readability):
```
C:\Users\Username\Documents\PowerShell\Modules;
C:\Program Files\PowerShell\Modules;
c:\program files\powershell\6.0.2\Modules;
C:\Program Files\WindowsPowerShell\Modules;
C:\MyVeryOwnModules;
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;
C:\Users\Username\Documents\WindowsPowerShell\Modules
```

Reply:
In that case.. we should split $WindowsPSModulePath and then do the check on all it's composite parts.

```powershell
    $Paths = @(
        "${env:userprofile}\Documents\WindowsPowerShell\Modules"
        "${env:programfiles}\WindowsPowerShell\Modules"
        [System.Environment]::GetEnvironmentVariable("psmodulepath", [System.EnvironmentVariableTarget]::Machine) -split [System.IO.Path]::PathSeparator
    )
    foreach ($Path in $Paths) {
        if (-not ($env:PSModulePath).Contains($Path))
        {
            $env:PSModulePath += ";$Path"
        }
    }
```

Reply:
I'm trying to address the issues of 1) order 2) ensuring the program files and user profile paths are added to the env var if missing from the computer env var 3) not repeating paths. I thin the code above addresses all 3.

Reply:
Yes, that would be more elegant.